From 255069fd26e800c65daae111579ad53d450f77ca Mon Sep 17 00:00:00 2001
From: Jason Wang <jasowang@redhat.com>
Date: Thu, 9 Oct 2014 06:13:23 -0400
Subject: [virt] hyperv: Remove unnecessary comparison of unsigned against 0

Message-id: <1412835210-22435-5-git-send-email-jasowang@redhat.com>
Patchwork-id: 97286
O-Subject: [RHEL7.1 PATCH V2 04/11] hv: Remove unnecessary comparison of unsigned against 0
Bugzilla: 1146358
RH-Acked-by: Vitaly Kuznetsov <vkuznets@redhat.com>

From: Tobias Klauser <tklauser@distanz.ch>

Bugzilla: 1146358

pfncount is of type u32 and thus can never be smaller than 0.

Found by the coverity scanner, CID 143213.

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>
Signed-off-by: K. Y. Srinivasan <kys@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
(cherry picked from commit 24b8a406bf5c3675bd193069cc6ab84a66fcbfd7)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/hv/channel.c b/drivers/hv/channel.c
index d90849c..da2bbfa 100644
--- a/drivers/hv/channel.c
+++ b/drivers/hv/channel.c
@@ -723,8 +723,7 @@ int vmbus_sendpacket_multipagebuffer(struct vmbus_channel *channel,
  u32 pfncount = NUM_PAGES_SPANNED(multi_pagebuffer->offset,
       multi_pagebuffer->len);
 
-
- if ((pfncount < 0) || (pfncount > MAX_MULTIPAGE_BUFFER_COUNT))
+ if (pfncount > MAX_MULTIPAGE_BUFFER_COUNT)
   return -EINVAL;
 
  /*
-- 
1.7.1